db-engine/src/main/java/com/splicemachine/db/impl/jdbc/EmbedPreparedStatement.java

```diff
@@ -1061,6 +1062,71 @@ public void addBatch() throws SQLException {
         }
     }
 
+    @Override
+    public int[] executeBatch() throws SQLException {
+        checkExecStatus();
+        synchronized (getConnectionSynchronization()) {
+            setupContextStack();
+            int i = 0;
+            // As per the jdbc 2.0 specs, close the statement object's current resultset
+            // if one is open.
+            // Are there results?
+            // outside of the lower try/finally since results will
+            // setup and restore themselves.
+            clearResultSets();
+
+            Vector stmts = batchStatements;
+            batchStatements = null;
+            int size;
+            if (stmts == null)
+                size = 0;
+            else
+                size = stmts.size();
+
+            int[] returnUpdateCountForBatch = new int[size];
+
+            SQLException sqle;
+            Iterator it = stmts.iterator();
+            activation.setBatch(it, this);
+
+            try {
+                activation.nextBatchElement();
+
+                returnUpdateCountForBatch = new int[size];
+                int r = 0;
+                do {
+                    if (super.executeStatement(activation, false, true))
+                        throw newSQLException(SQLState.RESULTSET_RETURN_NOT_ALLOWED);
+                    for (int j = 0; j < updateCounts.length; ++j) {
+                        returnUpdateCountForBatch[r++] = (int) updateCounts[j];
+                    }
+                } while(activation.nextBatchElement()); // if we are running on Spark we don't do batching, iterate over the elements here
+
+                InterruptStatus.restoreIntrFlagIfSeen(lcc);
+                return returnUpdateCountForBatch;
+            } catch (StandardException se) {
+
+                sqle = handleException(se);
+            } catch (SQLException sqle2) {
+                sqle = sqle2;
+            } finally {
+                activation.setBatch(null, null); // reset the batch state
+                restoreContextStack();
+            }
+
+            int successfulUpdateCount[] = new int[i];
```

Comment:
I think "I" here should be changed to "r".

Reply:
Currently for batched execution we fail all statements, so I've set this explicitly to an empty array.